include/turtle/detail/mutex.hpp

```diff
@@ -65,24 +74,30 @@ namespace mock
 {
 namespace detail
 {
-    struct mutex
+    struct mutex: boost::noncopyable
     {
-        mutex()
-        {}
-        void lock()
-        {}
-        void unlock()
-        {}
+        void lock(){}
+        void unlock(){}
     };
-    struct scoped_lock
+    // Dummy lock classes.
+    // Note: Ctor + Dtor make it RAII classes for compilers and avoid unused variable warnings
```

Comment:
nit: `Ctor` -> `Constructor` and `Dtor` -> `Destructor`, also `Note:` does not bring much (it's a comment, which by definition is sort of a note in the code 😄 )